src/chunk_pointer.rs

```diff
@@ -0,0 +1,35 @@
+use std::{
+    fs::File,
+    io::{self},
+    os::unix::fs::FileExt,
+    path::PathBuf,
+};
+
+#[allow(dead_code)]
+pub trait ChunkPointer {
+    fn read(&self) -> io::Result<Vec<u8>>;
+    fn new(path: PathBuf, offset: u64, size: usize) -> Self;
+}
+
+#[derive(Clone, Default)]
+pub struct ChunkHandler {
+    path: PathBuf,
+    offset: u64,
+    size: usize,
+}
+
+impl ChunkPointer for ChunkHandler {
+    fn new(path: PathBuf, offset: u64, size: usize) -> Self {
+        ChunkHandler { path, offset, size }
+    }
+
+    fn read(&self) -> io::Result<Vec<u8>> {
+        let file = Result::expect(File::open(self.path.clone()), "file error");
```

Comment:
в целом главу почитай https://doc.rust-lang.org/book/ch09-00-error-handling.html
```suggestion
        let file = File::open(self.path.clone()).expect("file error");
```